Refactor WebController

RickWieman opened the pull request:
Move pretty String builder of randomization to the RandomColorState.
Add toplevel commands without id (e.g. /random instead of just /random/{id})

Commit:
Refactor WebController

Add toplevel commands without id (e.g. /random instead of just /random/{id})
Move pretty String builder for lamps to a static method in WebController

Commit:
Move Philips Hue related stuff to a facade

Add an interface for Hue operations
Move "low-level code" to Hue facade
Remove Philips Hue imports from HueService
Add javadoc to HueService
Change /color endpoints to new pretty format
Update integration test

Commit:
Repair index page

Introduce HueLamp object to maintain facade
Add possibility to get all lamps (id and name) from HueService

praseodym commented:
Looking good! I only have some doubts about the `PHBridge` return object in the facade. The facade is now still hard-coupled to the Philips Hue API, so we cannot just provide a dummy implementation. It is probably hard to refactor this (we'd need many more facade methods), but it might give us a nicer overall architecture.

Commit:
Remove PHBridge from HueCommands

Replace getBridge by updateLightState (the only used method of the bridge)

RickWieman commented:
I replaced/removed it in cdc241d :smiley:

praseodym commented on src/main/java/ch/wisv/chue/hue/CHueFacade.java:
```diff
@@ -0,0 +1,149 @@
+import java.util.stream.Collectors;
+
+@Component
+public class CHueFacade implements HueFacade {
```
This should be named `PhilipsHueFacadeImpl` or something more descriptive

Commit:
Remove PHLightState from HueCommands

Add own implementation: HueLightState
Add logging facade (dummy)
Rename CHueFacade to PhilipsHueFacade
Make use of Optionals for syntactical reasons

Commit:
Fully replace hue/saturation/brightness

Rely on javax Color
Remove some obsolete comments

praseodym commented:
Looking good! Haven't had time to test it yet though.

RickWieman commented:
Tested at CH, seems to work. :shipit:?

praseodym commented on src/main/java/ch/wisv/chue/hue/HueLightState.java:
```diff
@@ -0,0 +1,108 @@
+    private OptionalInt transitionTime = OptionalInt.empty();
+    private Optional<Color> color = Optional.empty();
+
+    protected static PHLightState asPHLightState(HueLightState hueLightState) {
```
This class shouldn't depend on `com.philips.lighting.*`, this method can be moved to `PhilipsHueFacade`.

praseodym commented on src/main/java/ch/wisv/chue/hue/HueLightState.java:
```diff
@@ -0,0 +1,108 @@
+    @Override
+    public String toString() {
+        return "HueLightState{" +
+                "color=" + color +
```
This doesn't print anything useful

Commit:
Move LightState conversion to updateLightState

Commit:
Fix RGB rounding error

Make our RGB functions equal to Color.toString
